test/internal/options.go

```diff
@@ -0,0 +1,263 @@
+
+		if args["command"] == "jobdw" || args["command"] == "create_persistent" {
+			if name, found := args["profile"]; found {
+				t.options.storageProfile = &TStorageProfile{name: name}
+				return t.WithLabels("storage_profile")
+			}
+		}
+	}
+
+	panic(fmt.Sprintf("profile argument required but not found in test '%s'", t.Name()))
+}
+
+type TPersistentLustre struct {
+	name string
+
+	// Use internal tests to drive the persistent lustre workflow
+	create  *T
+	destroy *T
+
+	fsName  string
+	mgsNids string
+}
+
+func (t *T) WithPersistentLustre(name string) *T {
+	t.options.persistentLustre = &TPersistentLustre{name: name}
+	return t.WithLabels("persistent", "lustre")
+}
+
+type TCleanupPersistentInstance struct {
+	name string
+}
+
+// AndCleanupPersistentInstance will automatically destroy the persistent instance. It is useful
+// if you have a create_persistent directive that you wish to destroy after the test has finished.
+func (t *T) AndCleanupPersistentInstance() *T {
+	for _, directive := range t.directives {
+		args, _ := dwdparse.BuildArgsMap(directive)
+
+		if args["command"] == "create_persistent" {
+			t.options.cleanupPersistent = &TCleanupPersistentInstance{
+				name: args["name"],
+			}
+
+			return t
+		}
+	}
+
+	panic(fmt.Sprintf("create_persistent directive required but not found in test '%s'", t.Name()))
+}
+
+type TGlobalLustre struct {
+	fsName    string
+	mgsNids   string
+	mountRoot string
+
+	in  string // Create this file prior copy_in
+	out string // Expect this file after copy_out
+
+	persistent *TPersistentLustre // If using a persistent lustre instance as the global lustre
+}
+
+func (t *T) WithGlobalLustre(mountRoot string, fsName string, mgsNids string) {
+	panic("reference to an existing global lustre instance is not yet supported")
+}
+
+// WithGlobalLustreFromPersistentLustre will create a global lustre file system from a persistent lustre file system
+func (t *T) WithGlobalLustreFromPersistentLustre(mountRoot string) *T {
+	if t.options.persistentLustre == nil {
+		panic("Test option requires persistent lustre")
+	}
+
+	t.options.globalLustre = &TGlobalLustre{
+		persistent: t.options.persistentLustre,
+		mountRoot:  mountRoot,
+	}
+
+	return t.WithLabels("global_lustre")
+}
+
+// Prepare a test with the programmed test options
+func (t *T) Prepare(ctx context.Context, k8sClient client.Client) error {
+	o := t.options
+
+	if o.storageProfile != nil {
+		// Clone the placeholder profile
+		placeholder := &nnfv1alpha1.NnfStorageProfile{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "placeholder",
+				Namespace: "nnf-system",
+			},
+		}
+
+		Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(placeholder), placeholder)).To(Succeed())
+
+		profile := &nnfv1alpha1.NnfStorageProfile{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      o.storageProfile.name,
+				Namespace: "nnf-system",
+			},
+		}
+
+		placeholder.Data.DeepCopyInto(&profile.Data)
+		profile.Data.Default = false
```

Comment:
you might need to pin it as well?

Reply:
No. Pinned is what is done after a profile is cloned for use